core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java

```diff
@@ -69,7 +69,7 @@ public InputAction(Builder builder) {
         this.message = builder.message;
         this.variable = builder.variable;
         this.validAnswers = builder.validAnswers;
-        this.inputReader = Optional.ofNullable(builder.inputReader).orElse(new BufferedReader(new InputStreamReader(System.in)));
+        this.inputReader = Optional.ofNullable(builder.inputReader).orElseGet(() -> new BufferedReader(new InputStreamReader(System.in)));
```

Comment:
@christophd out of curiosity. how does citrus make sure this `inputReader` will be closed? I don't see it.

Reply:
@bbortt I also do not see it being closed properly

Reply:
:octocat: @picimako would you be willing to take a look at this, too? please?

Reply:
@bbortt Sure, I can look into it, and see what I can find.

Reply:
I think it's as easy as using a "try-with-resources" statement here:

```shell
diff --git a/core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java b/core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java
index 438310067..ea2409ee0 100644
--- a/core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java
+++ b/core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java
@@ -63,7 +63,7 @@ public class InputAction extends AbstractTestAction {
     /**
      * Default constructor.
      */
-    public InputAction(Builder builder) {
+    private InputAction(Builder builder) {
         super("input", builder);

         this.message = builder.message;
@@ -90,12 +90,9 @@ public class InputAction extends AbstractTestAction {
             display = message;
         }

-
-        try {
+        try (BufferedReader stdin = getInputReader()) {
             do {
                 logger.info(display);
-
-                BufferedReader stdin = getInputReader();
                 input = stdin.readLine();
             } while (validAnswers != null && !checkAnswer(input));
         } catch (IOException e) {
```

Reply:
@bbortt Yeah, that should do it.

I also did a quick search, and I see that the same problem should be fixed in `org.citrusframework.validation.script.TemplateBasedScriptBuilder#build` too. If you have no objection, I'll open a separate PR for these two changes.

Reply:
+1 on a separate PR for this. Many thanks!